pkg/tsdb/elasticsearch/response_parser.go

```diff
@@ -41,21 +44,27 @@ const (
 
 var searchWordsRegex = regexp.MustCompile(regexp.QuoteMeta(es.HighlightPreTagsString) + `(.*?)` + regexp.QuoteMeta(es.HighlightPostTagsString))
 
-func parseResponse(ctx context.Context, responses []*es.SearchResponse, targets []*Query, configuredFields es.ConfiguredFields, logger log.Logger) (*backend.QueryDataResponse, error) {
+func parseResponse(ctx context.Context, responses []*es.SearchResponse, targets []*Query, configuredFields es.ConfiguredFields, logger log.Logger, tracer tracing.Tracer) (*backend.QueryDataResponse, error) {
 	result := backend.QueryDataResponse{
 		Responses: backend.Responses{},
 	}
 	if responses == nil {
 		return &result, nil
 	}
+	ctx, span := tracer.Start(ctx, "datasource.elastic.parseResponse")
+	span.SetAttributes("responseLength", len(responses), attribute.Key("responseLength").Int(len(responses)))
 
 	for i, res := range responses {
+		_, span := tracer.Start(ctx, "datasource.elastic.parseResponse.response")
+		span.SetAttributes("queryMetricType", targets[i].Metrics[0].Type, attribute.Key("queryMetricType").String(targets[i].Metrics[0].Type))
 		start := time.Now()
 		target := targets[i]
 
 		if res.Error != nil {
 			mt, _ := json.Marshal(target)
 			me, _ := json.Marshal(res.Error)
+			span.RecordError(errors.New(string(me)))
+			span.SetStatus(codes.Error, string(me))
```

Comment:
What do you think about renaming this span to something else? Just to not be confused by the span variable outside of the loop.

Reply:
Good idea! Renamed in https://github.com/grafana/grafana/pull/74750/commits/1509a01bb63bc85fccb069bdb2350ed20475f40a.